src/Native/System.Native/pal_networking.cpp

```diff
@@ -213,12 +216,33 @@ extern "C" int32_t IPv4StringToAddress(const uint8_t* address, uint8_t* buffer,
     return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
 }
 
+static void AppendScopeIfNecessary(uint8_t* string, int32_t stringLength, uint32_t scope)
```

Comment:
This is just to be consistent with the existing typing, esp. since the input is a manged string buffer. Regardless, I still think that `uint8_t` is the correct type: our strings are not `char` strings: they're UTF-8 strings, which are made up of bytes, not characters (i.e. indexing into a string does not necessarily produce a valid code unit). I think that typing our strings as `char*`s, etc. is disingenuous.